File: src/utils/db_utils.js
// EXTERNAL LIBRARY //
const sqlite3 = require('sqlite3').verbose(); // verbose 사용 이유 없음
const path = require('path'); 

// LOCAL VARAIBLE //
const dbFile = (path.join(__dirname, path.sep+'test.db').replace(path.sep+'app.asar', '').replace('\\src\\utils\\','/')).replace('\\','/');
console.log(dbFile)



function db_create() {
    let db = new sqlite3.Database('./test.db')
    // \(백슬래시) 추가 줄바꿈 ERROR 발생하는지 확인필요.
    db.run("create TABLE process \
    (id integer primary key AUTOINCREMENT , name VARCHAR(255), start_time time, end_time time, count INTEGER, date date)")
}

function db_conn() {
    let db = new sqlite3.Database(`${dbFile}`, sqlite3.OPEN_READWRITE, (err) => {
        if (err) {
            //console.error(err.message);
        } else {
            //console.log('Connected to the mydb database.');
        }
    });
    return db
}

function db_disconn(db) {
    db.close((err) => {
        if (err) {
          return console.error(err.message);
        }
        //console.log('Close the database connection.');
      });
    
}

async function db_comm(db,type,query=''){
    if (type == 'INSERT'){
        db.serialize(()=> {
            db.run(query)
        })
    }
    if (type == 'SELECT'){
        return new Promise(function(resolve,reject){
            db.all(query, function(err,rows){
                if(err){return reject(err);}
                resolve(rows);
              });
        })
    }
}

//db_create()

exports.db_conn = db_conn;
exports.db_disconn = db_disconn;
exports.db_comm = db_comm;

File: test/utils/db_utils.py
import sqlite3
#TODO connection pool 사용할것인지 고려 필요
class internal_DB(object):
    # DB 연결
    def __init__(self):
        self.conn = None
        self.Cur = None

    def create(self):
        pass
    def init(self,name='test'):
        self.conn = sqlite3.connect(f'./{name}.db')

        self.Cur = self.conn.cursor()

    def CREATE(self):
        # 1분간 눈깜빡임 | 연속그룹 | 날짜 | 측정 시간대
        try:
            sql = f'CREATE TABLE process(id INTEGER PRIMARY KEY, count1min Integer, groupid text, date text, start_time text, end_time text)'
            self.Cur.execute(sql)
        except Exception as e:
            print('error',e)
        finally:
            self.conn.commit()

    def SELECT(self,query=None):
        try:
            sql = f'select * from process'
            self.Cur.execute(sql)
        except Exception as e:
            print('error',e)
        finally:
            self.conn.commit()

    def INSERT(self,data):
        COLUMNS = [key for key in data.keys()]
        VALUES = [data[val] for val in COLUMNS] 
        COLUMNS = ','.join(COLUMNS)

        for idx in range(len(VALUES)):
            if type(VALUES[idx]) == str:
                VALUES[idx] = "\"" + VALUES[idx] + "\""
            elif type(VALUES[idx]) == int:
                VALUES[idx] = str(VALUES[idx])
        
        VALUES = ','.join(VALUES)
        #print(VALUES)

        try:
            sql = f'INSERT INTO process ({COLUMNS}) VALUES ({VALUES})'
            self.Cur.execute(sql)
        except Exception as e:
            print('error',e)
        finally:
            self.conn.commit()

#instance = internal_DB()
#instance.init('main')
#instance.CREATE()
#instance.SELECT()